shopfloor/services/checkout.py

```diff
@@ -620,16 +623,27 @@ def _select_lines_from_delivery_packaging(
                     packaging, carrier
                 ),
             )
+        message = None
+        need_confirm_pack_all = ""
+        lines_to_pack = selection_lines.filtered(self._filter_lines_to_pack)
+        if not lines_to_pack:
+            if self.work.menu.no_prefill_qty:
+                message = self.msg_store.no_lines_to_process_set_quantities()
+            elif not confirm_pack_all == packaging.barcode:
+                need_confirm_pack_all = packaging.barcode
+                message = self.msg_store.confirm_put_all_goods_in_delivery_package(
+                    packaging
+                )
+            if message:
+                return self._response_for_select_line(
+                    picking,
+                    message=message,
+                    need_confirm_pack_all=need_confirm_pack_all,
+                )
         if confirm_pack_all:
-            # Select all lines and pack them all w/o passing for select_package state
             self._select_lines(selection_lines)
```

Comment:
Should be the `else` condition in the above `if not lines_to_pack:` block

Reply:
That does not work, on a valid confirmation to pack everything there will still be nothing to pack !
I refined the if case to be more explicit.